File: nodejs/seguranca-ms/src/core/user/dto/create-user-request.ts
import {
  IsEmail,
  IsEnum,
  IsNotEmpty,
  Length,
  MaxLength,
} from 'class-validator';
import { ApiProperty } from '@nestjs/swagger';
import { UserProfile } from './user-profile.enum';

export class CreateUserRequest {
  @IsNotEmpty({
    message: "o atributo 'name' deve ser informado.",
  })
  @Length(5, 50, {
    message:
      "O atributo 'name' deve possuir no mínimo 5 e no máximo 50 caracateres.",
  })
  @ApiProperty()
  name: string;

  @IsNotEmpty({
    message: "o atributo 'email' deve ser informado.",
  })
  @MaxLength(100, {
    message: "O atributo 'email' deve possuir no máximo 100 caracateres.",
  })
  @IsEmail()
  @ApiProperty()
  email: string;

  @IsNotEmpty({
    message: "o atributo 'login' deve ser informado.",
  })
  @Length(5, 30, {
    message:
      "O atributo 'login' deve possuir no mínimo 5 e no máximo 30 caracateres.",
  })
  @ApiProperty()
  login: string;

  @IsEnum(UserProfile, {
    message: "O atributo 'profile' deve ser informado com um dos seguintes valores [" + Object.values(UserProfile) + "]",
  })
  @ApiProperty()
  profile: UserProfile;
}

File: java/quarkus/seguranca-ms/src/main/java/br/ejcb/cfp/seguranca/application/dto/user/CreateUserRequest.java
package br.ejcb.cfp.seguranca.application.dto.user;

import br.ejcb.cfp.seguranca.common.enumeration.UserProfile;
import jakarta.validation.constraints.Email;
import jakarta.validation.constraints.NotEmpty;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Size;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.With;

@Getter
@Setter
@With
@NoArgsConstructor
@AllArgsConstructor
public class CreateUserRequest {

	@NotEmpty(message = "O atributo 'name' deve ser informado")
	@Size(min = 10, max = 100, message = "O atributo 'name' deve ser preenchido e possuir um tamanho entre 10 e 100 caracteres")
	private String name;

	@NotEmpty(message = "O atributo 'email' deve ser informado")
	@Email(message = "'email' informado é inválido")
	private String email;

	@NotEmpty(message = "O atributo 'login' deve ser informado")
	@Size(min = 10, max = 50, message = "O atributo 'login' deve ser preenchido e possuir um tamanho entre 10 e 50 caracteres")
	private String login;

	@NotNull(message = "O atributo 'profile' deve ser informado")
	private UserProfile profile;
	
}
